javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,264 @@
+    }
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+function getNote(id) {
+    for (const note of notes) {
+        if (note.id === id) {
+            return note;
+        }
+    }
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+function logOutNotesFormatted() {
+    for (const note of notes)
+        console.log(
+            `The note with id: ${note.id}, has the following note text: ${note.content}`
+        );
+}
+
+logOutNotesFormatted(); // should log out the text below
+
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+// This part of the code is copied from: https://codepen.io/dalsHughes/pen/poJGejX
+const textarea = document.querySelector("textarea");
+const ul = document.querySelector("ul");
+let noteId = 0;
+
+document.querySelector("button.add-note").addEventListener("click", () => {
+    saveNote(textarea.value, noteId);
+    noteId++;
+    textarea.value = "";
+
+    // Clear the ul
+    ul.innerHTML = "";
+    notes.forEach((note) => {
+        const li = document.createElement("li");
+        li.innerHTML = note.content;
+
+        // ! ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+        // I think these would make the app better,
+        // if a note is important, the user can make its background red by clicking on it,
+        // also if a note is done, the user by double-clicking on it will make it line-through
+
+        // To make a note Important by adding a class with red background color
+        li.onclick = () => li.classList.toggle("isImportant");
+
+        // To make a note Done by adding a class with line-through style
+        li.ondblclick = () => li.classList.toggle("isDone");
+        // ! ↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑
+
+        ul.appendChild(li);
+    });
+});
+
+document
+    .querySelector("button.log-out")
+    .addEventListener("click", logOutNotesFormatted);
+
+console.log(
+    "-----------------------------------------------------------------"
+);
+
+// CactusIO-interactive (Smart phone usage app) optional
+
+// Adding an activity
+
+// Initializing an empty array for restoring the activities
+let activities = [];
+
+// To get the activity day in this format: 2/30/2024
+const today = new Date();
+const activityDay = today.toLocaleDateString();
+
+// To create an object from any activity and then adding it to the activities array
+function addActivity(activity, duration) {
+    activities.push({
+        date: activityDay,
+        activity: activity,
+        duration: duration,
+    });
+}
+
+addActivity("Youtube", 30);
+addActivity("Instagram", 40);
+addActivity("Facebook", 8);
+addActivity("Gaming", 20);
+
+// Show my status
+
+// Default value for usage limit
+let usageLimit = 90;
```

Comment:
There is a general practice of defining all variables in the top of the files where they are needed, or if the scope is more private - in the top of the related function. In this case, if I were you, I'd define `usageLimit` after `activityDay`.
Also this can be `const`, unless you want to expand the code and have a different limit depending on the current day being a week or weekend ;)